src/Bathymetry.jl

```diff
@@ -172,15 +154,23 @@ function regrid_bathymetry(target_grid;
         z_data[land] .= height_above_water
     end
 
-    # Build the "native" grid of the bathymetry and make a bathymetry field.
+    # Infer the "native grid" of the bathymetry data and make a bathymetry field.
+    Δλ = λ_data[2] - λ_data[1]
+    Δφ = φ_data[2] - φ_data[1]
+
+    λ₁_data = λ_data[i₁] - Δλ / 2
+    λ₂_data = λ_data[i₂] + Δλ / 2
+    φ₁_data = φ_data[j₁] - Δφ / 2
+    φ₂_data = φ_data[j₂] + Δφ / 2
+
     Nxn = length(λ_data)
     Nyn = length(φ_data)
     Nzn = 1
 
     native_grid = LatitudeLongitudeGrid(arch;
                                         size = (Nxn, Nyn, Nzn),
-                                        latitude = (φ₁, φ₂),
-                                        longitude = (λ₁, λ₂),
+                                        latitude = (φ₁_data, φ₂_data),
+                                        longitude = (λ₁_data, λ₂_data),
```

Comment:
@simone-silvestri please inform whether this is correct.

Reply:
should be correct